.github/workflows/check-generated-diff.yml

```diff
@@ -0,0 +1,60 @@
+name: Validate Generated Files
+on:
+  pull_request:
+    paths:
+      - 'backend/api/**/*.proto'
+      - '.github/workflows/validate-proto-gen.yml'
+      - 'backend/api/**/go_http_client/**'
+      - 'backend/api/**/go_client/**'
+      - 'backend/api/**/python_http_client/**'
+      - 'backend/api/**/swagger/**'
+
+jobs:
+  validate-generated-files:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        api_version: ['v1beta1', 'v2beta1']
+      fail-fast: false
+
+    container:
+      image: golang:1.21
```

Comment:
can we synchronize the go version with the project go.mod? 

also is it necessary we use a go container here? could we instead use something like this: 

```
    - name: Install Go
      uses: actions/setup-go@v5
      with:
        go-version-file: go.mod
```

like we do [here](https://github.com/kubeflow/pipelines/blob/e89d2d5f2d4835d92c2e5739f0a12c20f8106017/.github/workflows/backend.yml#L26C1-L29C32), this would keep the go version always in sync with the project's